.github/workflows/ci.yml

```
@@ -79,20 +108,20 @@ jobs:
         run: ctest --preset=core-test-${{ matrix.platform }}
 
       - name: Zip output folder
-        if: startsWith(github.ref, 'refs/tags/')
+        if: startsWith(github.ref, 'refs/tags/') && (matrix.platform != 'linux' || matrix.compiler == 'gcc-default')
```

Should it be `matrix.platform == 'linux'`, or am I being dumb? I'm not sure I completely understand these lines.

---

It's not obvious... I should probably add a comment.

This step and the one after it are only run for new releases. They zip up the binaries then upload them to GitHub as release artifacts. We want to run this step for every platform, but we only want to upload one set of binaries for Linux, even though we're building it several times with different compilers.

---

I think that mixing platforms and compiles muddy the explanation a bit. Maybe use, so you choose what to upload based on the compiler only.

```suggestion
        if: startsWith(github.ref, 'refs/tags/') && (matrix.compiler == 'msvc' || matrix.compiler == 'gcc-default')
```

---

This would work now, but won't work when I add macOS to the runners. I agree this is a bit ugly though. How about I add another parameter which indicates whether the compiler is the "main" one or not (e.g. a `matrix.is_main` boolean value you could check instead)?

---

Sounds like a good option